### PR TITLE
test(e2e): initial framework

### PR DESCRIPTION
**Pull request by adamreese**
See https://github.com/kubernetes/helm/pull/2846

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@adamreese Is there any manual validation that should be done on this? Based on what I can see, the tests running in circle should just cover it. I'll be going over all of the files tomorrow for a full review

**Comment by adamreese** [comment]
@thomastaylor312 Don't worry about a full review on this.  I am going to play with `minikube --vm-driver=none` to see how that could simplify the setup.

**Comment by michelleN** [comment]
Hey this is pretty dope. I see we're building our own kubernetes cluster here to text against. Would it possible or even a good idea to add an option to test again the current kubernetes cluster I have running in minikube instead?

File: Makefile [hunk 1]
```diff
@@ -117,6 +117,18 @@ clean:
...
+dind: docker-build
+dind:
+	@scripts/portforward.sh start
+	@scripts/dind.sh
+	@scripts/import-docker-image.sh
+
+.PHONY: e2e
+e2e: dind
+e2e:
+	go test -v ./e2e --cluster-url http://localhost:8080 --helm-bin ../bin/helm
```
Comment (thomastaylor312) [inline Makefile]: The cluster URL should probably use variables (set with this default value) so you can override them if testing against another cluster of your choice

File: e2e/e2e_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
```
Comment (thomastaylor312) [inline e2e/e2e_suite_test.go]: Nit: Copyright should be 2018. Same goes for a bunch of the other files

File: glide.yaml [hunk 3]
```diff
@@ -53,15 +53,15 @@ import:
   vcs: git
 
 - package: k8s.io/kubernetes
-  version: 1.9.2
+  version: 1.9.3
```
Comment (thomastaylor312) [inline glide.yaml]: This will need to be rebased against master with 1.10 support

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Few small discussion points. The only big thing is that this will need to be rebased against the latest changes with k8s 1.10

**Comment by adamreese** [comment]
Lets abort this and revisit for helm v3.

/close

**Pull request closed without merging**